src/main/java/org/openpnp/gui/VisionSettingsPanel.java

```diff
@@ -0,0 +1,203 @@
+        toolbarPanel.add(toolBar);
+
+        JPanel upperPanel = new JPanel();
+        toolbarPanel.add(upperPanel, BorderLayout.EAST);
+
+        toolBar.add(newSettingsAction);
+        toolBar.add(deleteSettingsAction);
+
+        toolBar.addSeparator();
+    }
+
+    private AbstractVisionSettings getSelection() {
+        List<AbstractVisionSettings> selections = getSelections();
+        if (selections.size() != 1) {
+            return null;
+        }
+        return selections.get(0);
+    }
+
+    private List<AbstractVisionSettings> getSelections() {
+        List<AbstractVisionSettings> selections = new ArrayList<>();
+        for (int selectedRow : table.getSelectedRows()) {
+            selectedRow = table.convertRowIndexToModel(selectedRow);
+            selections.add(tableModel.getVisionSettings(selectedRow));
+        }
+        return selections;
+    }
+
+    public final Action newSettingsAction = new AbstractAction() {
+        {
+            putValue(SMALL_ICON, Icons.add);
+            putValue(NAME, "New Settings...");
+            putValue(SHORT_DESCRIPTION, "Create a new settings, specifying it's ID.");
+        }
+
+        @Override
+        public void actionPerformed(ActionEvent e) {
+            String id;
+            while ((id = JOptionPane.showInputDialog(frame,
+                    "Please enter an ID for the new settings.")) != null) {
+                if (Configuration.get().getBottomVisionSettings(id) == null) {
+                    AbstractVisionSettings visionSettings = new BottomVisionSettings(id);
+
+                    Configuration.get().addVisionSettings(visionSettings);
+                    tableModel.fireTableDataChanged();
+                    Helpers.selectLastTableRow(table);
+                    break;
+                }
+
+                MessageBoxes.errorBox(frame, "Error", "VisionSettings ID " + id + " already exists.");
+            }
+        }
+    };
+
+    public final Action deleteSettingsAction = new AbstractAction() {
+        {
+            putValue(SMALL_ICON, Icons.delete);
+            putValue(NAME, "Delete Settings");
+            putValue(SHORT_DESCRIPTION, "Delete the currently selected settings.");
+        }
+
+        @Override
+        public void actionPerformed(ActionEvent e) {
+            List<AbstractVisionSettings> selections = getSelections();
+
+            List<String> usedIn = new ArrayList<>();
+            for (AbstractVisionSettings settings : selections) {
+                for (Package pkg : Configuration.get().getPackages()) {
+                    if (pkg.getVisionSettings() == settings) {
+                        usedIn.add(pkg.getId());
+                    }
+                }
+
+                for (Part part : Configuration.get().getParts()) {
+                    if (part.getVisionSettings() == settings) {
+                        usedIn.add(part.getId());
+                    }
+                }
+            }
+
```

Comment:
Should this also check against the BottomVision default?

Reply:
What do you mean?

Reply:
The Settings could be referenced here:
https://github.com/openpnp/openpnp/blob/a065d93a5987a3c7bf1963a62e51dd2dd30fd341/src/main/java/org/openpnp/machine/reference/vision/ReferenceBottomVision.java#L34
If yes, it should not allow to delete them.

Reply:
I'm still not sure I follow, sorry 😄 
So you mean to have the reference of used `BottomVissionSettings` in the `ReferenceBottomVision` instead of looping through all the parts and packages to check the usage right?

Reply:
Maybe I'm missing something. But from what I understand in the code, a user could delete the default  `BottomVisionSettings`, the one that is assigned here (if it is not the same time assigned to any part or package):

https://github.com/openpnp/openpnp/blob/a065d93a5987a3c7bf1963a62e51dd2dd30fd341/src/main/java/org/openpnp/machine/reference/vision/ReferenceBottomVision.java#L34

Note, when inheritance is active, the setting is `null`, i.e. the inheritance is _not_ evaluated in the in-use check loop above (and that's OK).

Reply:
User cannot delete this default settings. It is the build-in one that is immutable. User can delete only the custom ones. The delete is performed upon the selection in the table of VisionSettings and the default settings in not present in that table.